aten/src/ATen/native/cuda/ReplicationPadding.cu

```diff
@@ -184,26 +311,406 @@ void replication_pad1d_backward_out_cuda_template(
+  int owidth  = iwidth + pleft + pright;
+  AT_CHECK(owidth >= 1 || oheight >= 1 || odepth >= 1,
+      "input (D: ", idepth, " H: ", iheight, ", W: ", iwidth,
+      ") is too small."
+      " Calculated output D: ", odepth, " H: ", oheight, " W: ", owidth);
+
+  AT_CHECK(at::cuda::detail::canUse32BitIndexMath(gradOutput),
+      "output gradient tensor must fit into 32-bit index math");
+
+  AT_CHECK(numPlanes == gradOutput.size(planeDim),
+      "gradOutput width unexpected. Expected: ", numPlanes, ", Got: ",
+      gradOutput.size(planeDim));
+  AT_CHECK(owidth == gradOutput.size(dimw),
+      "gradOutput width unexpected. Expected: ", owidth, ", Got: ",
+      gradOutput.size(dimw));
+  AT_CHECK(oheight == gradOutput.size(dimh),
+      "gradOutput height unexpected. Expected: ", oheight, ", Got: ",
+      gradOutput.size(dimh));
+  AT_CHECK(odepth == gradOutput.size(dimd),
+      "gradOutput depth unexpected. Expected: ", odepth, ", Got: ",
+      gradOutput.size(dimd));
+}
+
+void replication_pad3d_out_cuda_template(
+    Tensor& output,
+    const Tensor& input,
+    IntList paddingSize)
+{
+  AT_CHECK(paddingSize.size() == 6, "padding Size is expected to be 6");
+  int pleft = paddingSize[0];
+  int pright = paddingSize[1];
+  int ptop = paddingSize[2];
+  int pbottom = paddingSize[3];
+  int pfront = paddingSize[4];
+  int pback = paddingSize[5]; 
+  shapeCheck3d(input, pleft, pright, ptop,
+      pbottom, pfront, pback);
+
+  int planeDim = 0;
+  int dimd = 1;
+  int dimh = 2;
+  int dimw = 3;
+  int numBatch = 1;
+
+  int numInputDims = input.dim();
+
+  if (numInputDims == 5) {
+    numBatch = input.size(0);
+    planeDim++;
+    dimd++;
+    dimh++;
+    dimw++;
+  }
+
+  int numPlanes = input.size(planeDim);
+  int inputD = input.size(dimd);
+  int inputH = input.size(dimh);
+  int inputW = input.size(dimw);
+  int outputD = inputD + pfront + pback;
+  int outputH = inputH + ptop + pbottom;
+  int outputW  = inputW + pleft + pright;
+
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(
+      input.type(), "replication_pad3d", [&] {
+
+      if (numInputDims == 4) {
+      output.resize_({numPlanes, outputD, outputH, outputW});
+      auto input_ = input.reshape({1, input.size(0), input.size(1), input.size(2), input.size(3)});
+      auto output_ = output.reshape({1, output.size(0), output.size(1), output.size(2), output.size(3)});
+      auto devInput = input_.packed_accessor<scalar_t, 5>();
+      auto devOutput = output_.packed_accessor<scalar_t, 5>();
+
+      int outputPlaneSize = devOutput.size(2) * devOutput.size(3) *
+      devOutput.size(4);
+      dim3 gridSize(THCCeilDiv(outputPlaneSize, 256),
+          devOutput.size(1),
+          devOutput.size(0));
+      dim3 blockSize(outputPlaneSize > 256 ? 256 : outputPlaneSize);
+
+      replication_pad_forward_kernel3d <<<gridSize, blockSize, 0,
+      at::cuda::getCurrentCUDAStream()>>>(
+          devInput, devOutput, pfront, pback, ptop, pbottom, pleft, pright);
```

Comment:
I did not see an error, but it was pretty disconcerting to see `pfront` and `pback` suddenly show up at the beginning of the argument list, even though they were in the back of paddingSize